python/pyspark/sql/conversion.py

```diff
@@ -0,0 +1,543 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from pyspark.sql.pandas.utils import require_minimum_pyarrow_version
+
+# Arrow is required for arrow conversion.
+require_minimum_pyarrow_version()
+
+
+import array
+import datetime
+import decimal
+from typing import Any, Callable, List, Sequence
+
+import pyarrow as pa
```

Comment:
Still Arrow is an optional package. Without installing it, it fails when you import PySpark without Arrow.

Reply:
You will have to import inplace where it's used, e.g., within the method.

Reply:
We have called `pyspark.sql.pandas.utils.require_minimum_pyarrow_version()` before this import to make sure arrow is present.

Reply:
`require_minimum_pyarrow_version` throws an exception, and `pyspark.sql` doesn't require PyArrow by default. If this package is imported somewhere within `pyspark.sql`, it will fail to import `pyspark.sql.` at all.

Reply:
Changed to only import where it's used